authz/rbac_translator.go

```diff
@@ -266,6 +281,69 @@ func parseRules(rules []rule, prefixName string) (map[string]*v3rbacpb.Policy, e
 	return policies, nil
 }
 
+// Parse auditLoggingOptions to the associated RBAC protos. The single
+// auditLoggingOptions results in two different parsed protos, one for the allow
+// policy and one for the deny policy
+func (options *auditLoggingOptions) toProtos() (allow *v3rbacpb.RBAC_AuditLoggingOptions, deny *v3rbacpb.RBAC_AuditLoggingOptions, err error) {
+	allowRbac := v3rbacpb.RBAC_AuditLoggingOptions{}
+	denyRbac := v3rbacpb.RBAC_AuditLoggingOptions{}
+
+	if options.AuditCondition != "" {
+		rbacCondition, ok := v3rbacpb.RBAC_AuditLoggingOptions_AuditCondition_value[options.AuditCondition]
+		if !ok {
+			return nil, nil, fmt.Errorf("failed to parse AuditCondition %v. Allowed values {NONE, ON_DENY, ON_ALLOW, ON_DENY_AND_ALLOW}", options.AuditCondition)
+		}
+		allowRbac.AuditCondition = v3rbacpb.RBAC_AuditLoggingOptions_AuditCondition(rbacCondition)
+		denyRbac.AuditCondition = toDenyCondition(v3rbacpb.RBAC_AuditLoggingOptions_AuditCondition(rbacCondition))
+	}
+
+	for i := range options.AuditLoggers {
+		config := &options.AuditLoggers[i]
+		if config.Config == nil {
+			return nil, nil, fmt.Errorf("AuditLogger Config field cannot be nil")
+		}
+		customConfig, err := anypb.New(config.Config)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error parsing custom audit logger config: %v", err)
```

Comment:
Nit: does it make sense to include the erroring config in the error string?

Reply:
The custom config is a user-defined json object so it could be very large, would we want to write it out in the error message?